engine/gamesys/src/gamesys/components/comp_sprite.cpp

```diff
@@ -65,8 +65,27 @@ namespace dmGameSystem
         SpriteResourceOverrides() : m_Material(0) {}
     };
 
+    const uint32_t MAX_TEXTURE_COUNT = dmRender::RenderObject::MAX_TEXTURE_COUNT;
+
+    struct TexturesData
+    {
+        TextureSetResource*             m_Resources[MAX_TEXTURE_COUNT];
+        dmGameSystemDDF::TextureSet*    m_TextureSets[MAX_TEXTURE_COUNT];
+        uint32_t                        m_NumTextures;
+
+        // Used after resolving info from all textures
+        dmhash_t                        m_AnimationID;                      // The animation of the driving atlas
+        uint32_t                        m_Frames[MAX_TEXTURE_COUNT];        // The resolved frame indices
+        float                           m_PageIndices[MAX_TEXTURE_COUNT];
+
+        const dmGameSystemDDF::TextureSetAnimation* m_Animations[MAX_TEXTURE_COUNT];
+        const dmGameSystemDDF::SpriteGeometry*      m_Geometries[MAX_TEXTURE_COUNT];
+        bool                                        m_UsesGeometries;
+    };
+
     struct SpriteComponent
     {
+        TexturesData                m_TexturesData; // store precalculated gemetry data that used in different places during the frame
```

Comment:
Stores textures data to avoid double calculation in `UpdateTextureDataAndCounts` (ex `UpdateVertexAndIndexCount`) and in `CreateVertexData`. So calculate once in `UpdateTextureDataAndCounts` and use that result in `CreateVertexData`.

Reply:
I was a bit surprised to see this in a PR regarding the pivot point?

Also, I'd expect some numbers on the trade offs for this.
E.g. now the sprite component structure becomes much larger.
I.e. what was the size of 32k sprites before/after?
What is the _measured_ performance benefit.
I feel we need those things before making a judgement call, especially as we are currently looking to minimize runtime footprint where available.

Reply:
Adding those data to `SpriteComponent` is more about to solve current issue.
Pivots we calculate only inside `CreateVertexData`. Call chain: `RenderListDispatch`->`RenderBatch`->`CreateVertexData`. So `RenderListFrustumCulling` is calling before pivot points will be calculated.
So I guess more correct solution is to precalculate pivot points and stored it in `SpriteWorld` like as done for `m_BoundingVolumes`.